packages/context/src/resolution-session.ts

```diff
@@ -343,3 +339,22 @@ export interface ResolutionOptions {
    */
   optional?: boolean;
 }
+
+/**
+ * Resolution options or session
+ */
+export type ResolutionOptionsOrSession = ResolutionOptions | ResolutionSession;
+
+/**
+ * Normalize ResolutionOptionsOrSession to ResolutionOptions
+ * @param optionsOrSession resolution options or session
+ */
+export function asResolutionOptions(
+  optionsOrSession?: ResolutionOptionsOrSession,
+): ResolutionOptions {
+  // backwards compatibility
+  if (optionsOrSession instanceof ResolutionSession) {
+    optionsOrSession = {session: optionsOrSession};
```

Comment:
Can we avoid mutating arguments please?

```suggestion
    return {session: optionsOrSession};
```